compiler/rustc_borrowck/src/diagnostics/conflict_errors.rs

```diff
@@ -3828,11 +3828,16 @@ impl<'infcx, 'tcx> MirBorrowckCtxt<'_, 'infcx, 'tcx> {
             if tcx.is_diagnostic_item(sym::deref_method, method_did) {
                 let deref_target =
                     tcx.get_diagnostic_item(sym::deref_target).and_then(|deref_target| {
-                        Instance::try_resolve(tcx, self.param_env, deref_target, method_args)
-                            .transpose()
+                        Instance::try_resolve(
+                            tcx,
+                            self.infcx.typing_env(self.param_env),
```

Comment:
this is an interesting theoretical issue: we're using something taking a `TypingEnv` while defining opaque types. Given that to correctly do so we need to pass in the current opaque type storage (and be able to return constraints), I don't know how'll be able to *correctly* handle opaque types here. This is only for diagnostics so it shouldn't matter.